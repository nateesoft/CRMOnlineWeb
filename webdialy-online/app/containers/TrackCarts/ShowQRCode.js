import React, { useState} from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
const QRCode = require('qrcode.react');
import * as appConstants from 'containers/App/constants';

const ShowQRCode = props => {
  const { onClose, open, cart: cart_no, db: database } = props;
  const loc = window.location.href.split('/');
  const hostUrl = `${loc[0]}//${loc[2]}${appConstants.publicPath}`;

  return (
    <Dialog
      onClose={onClose}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
      open={open}
    >
      <DialogTitle id="simple-dialog-title">QR CODE</DialogTitle>
      <DialogContent>
          <DialogContentText id="alert-dialog-description">
            <QRCode value={`${hostUrl}/order_confirm/${cart_no}/${database}`} />
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={onClose} color="primary" autoFocus>
            Close
          </Button>
        </DialogActions>
    </Dialog>
  );
};

ShowQRCode.propTypes = {
  open: PropTypes.bool.isRequired,
};

export default ShowQRCode;
