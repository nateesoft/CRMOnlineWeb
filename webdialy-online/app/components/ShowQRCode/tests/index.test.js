import React from 'react';
import renderer from 'react-test-renderer';

import ShowQRCode from '../index';

describe('ShowQRCode components', () => {
  it('test snapshot', () => {
    const component = renderer.create(<ShowQRCode />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
