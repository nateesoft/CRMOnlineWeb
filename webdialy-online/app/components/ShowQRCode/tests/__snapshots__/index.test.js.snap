// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowQRCode components test snapshot 1`] = `null`;
